docs/config.rst

```diff
@@ -1,11 +1,505 @@
+If you then change your project's ``cumulusci.yml`` file to also specify a default value for o2, this will take precedence over the default value specified in your global ``cumulusci.yml`` file.
+
+The following diagram illustrates these three files along with their corresponding scopes in green.
+Changes made to configuration files on top will override any changes in files below them (if they are present).
+
+.. image:: images/cci-yml-chart.png
+
+
+
+Project Configurations
+^^^^^^^^^^^^^^^^^^^^^^
+**macOS/Linux:** ``.../path/to/project/cumulusci.yml``
+**Windows:** ``...\path\to\project\cumulusci.yml``
+
+This ``cumulusci.yml`` file lives in the root directory of your project, and apply to only this project.
+Changes here can be commited back to a remote repository so other team members can benefit from the customizations.
+Configurations in this file apply to this project, and take precedence over any configurations specified in the `global configurations`_ file, but are overridden by configurations in the `local project`_ file.
+
+
+
+Local Project Configurations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+**macOS/Linux:** ``~/.cumulusci/project_name/cumulusci.yml``
+**Windows:** ``%homepath%\.cumulusci\project_name\cumulusci.yml``
+
+Configurations made to this ``cumulusci.yml`` file apply to only the project with the given <project_name>, and take precedence over **all other** configuration scopes.
+If you want to make customizations to a project, but don't need them to be available to other team members, you would make those customizations here.
+
+
+
+Global Configurations
+^^^^^^^^^^^^^^^^^^^^^
+**macOS/Linux:** ``~/.cumulusci/cumulusci.yml``
+**Windows:** ``%homepath%\.cumulusci\cumulusci.yml``
+
+Configuration of this file will override behavior across **all** CumulusCI projects on your machine.
+Configurations in this file have the lowest precedence, and are overridden by **all other** configuration scopes.
+
+
+
+The Internal ``cumulusci.yml``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+There is one more configuration file that exists: the `internal cumulusci.yml <https://github.com/SFDO-Tooling/CumulusCI/blob/master/cumulusci/cumulusci.yml>`_ file that ships with CumulusCI itself.
+This file actually holds the lowest precedence of all, as all other scopes override what is contained in this file.
+This file contains all of the definitions for the standard tasks, flows, and org configurations that come out of the box with CumulusCI.
+As a CumulusCI user you aren't able to modify it, but knowing about it serves two purposes:
+
+* It is a fun bit of trivia to know that this file exists!
+* It is useful to reference when working on configuring custom tasks or flows of your own.
+
+
+Advanced Configurations
+-----------------------
+
+Using Variables for Task Options
```

Comment:
I'd be inclined to put this and the section about referencing task return values underneath the "Task Configurations" section, rather than under Advanced Configurations.